tests/unit/wc-payment-api/test-class-wc-payments-api-client.php

```diff
@@ -2079,6 +2081,29 @@ public function test_authorizations_summary_success() {
 		$this->assertSame( 1200, $summary['total'] );
 	}
 
+	public function test_wcpay_request_with_value_object() {
+		$expected_amount = 123;
+
+		$this->set_http_mock_response(
+			200,
+			[
+				'id'          => 'charge_id',
+				'customer_id' => 1,
+				'amount'      => $expected_amount,
+				'created'     => 1557224304,
+				'currency'    => 'usd',
+			]
+		);
+
+		$create_charge = new Create_Charge();
+		$create_charge
+			->set_amount( $expected_amount )
+			->set_source_id( 'source_id' );
+		$result = $this->payments_api_client->send_wcpay_request( $create_charge );
+		$this->assertInstanceOf( Response::class, $result );
```

Comment:
Kudos for providing this usage example! I love the direction the API is going!

I'd suggest taking bit more control over creation
```php
$create_charge = Create_Charge::new/make()
    ->chained_method()
    // ...
```

Reply:
Thank you for the suggestion. I will add it.

Reply:
Added.